### PR TITLE
Add spec identifier syntax to destructors.md

### PR DESCRIPTION
**Pull request by chorman0773**
This adds spec identifier syntax for destructors.md

### CODE REVIEW

File: src/destructors.md [hunk 1]
```diff
@@ -57,48 +62,85 @@ core::mem::forget(partial_move.1);
 
 ## Drop scopes
 
+r[destructor.scope]
```
Comment (traviscross) [inline src/destructors.md:65]:
```suggestion
r[destructors.scope]
```

The document goes back and forth between the plural and singular.  Seems these should all be plural.

File: src/destructors.md [hunk 2]
```diff
@@ -57,48 +62,85 @@ core::mem::forget(partial_move.1);
...
+
```
Comment (traviscross) [inline src/destructors.md:66]:
```suggestion
```

Let's remove the double line breaks.

File: src/destructors.md [hunk 3]
```diff
@@ -1,11 +1,15 @@
 # Destructors
+r[destructors]
```
Comment (traviscross) [inline src/destructors.md:2]: In another PR, where the `r[..]` had the same name as the primary header, it was elided.  Do we want to elide these or include them?

File: src/destructors.md [hunk 4]
```diff
@@ -346,6 +413,8 @@ let x = (&temp()).use_temp();  // ERROR
 
 ## Not running destructors
 
+r[destructors.forget]
+
```
Comment (traviscross) [inline src/destructors.md:417]: Let's be consistent on whether or not there's a line break between the `r[..]` tag and the content that follows when it's ahead of a paragraph like this.

**Review by traviscross — CHANGES_REQUESTED** [review]

File: src/destructors.md [hunk 5]
```diff
@@ -1,11 +1,15 @@
 # Destructors
+r[destructors]
 
+
+r[destructors.general]
```
Comment (traviscross) [inline src/destructors.md:5]:
```suggestion
r[destructors.intro]
```

In the other chapter, we went with `.intro` for cases like these.  The particular name probably doesn't matter, but we should be consistent.  `intro` is shorter, so maybe let's do that.

File: src/destructors.md [hunk 6]
```diff
@@ -57,48 +62,85 @@ core::mem::forget(partial_move.1);
 
 ## Drop scopes
 
+r[destructor.scope]
+
+
```
Comment (traviscross) [inline src/destructors.md:67]:
```suggestion

```

Let's remove the double line breaks here and elsewhere.

**Comment by traviscross** [comment]
We talked about this in our lang-docs call.  Other than as mentioned here, these changes seem OK to us.

File: src/destructors.md (hunk 4 above)
Comment (chorman0773) [reply src/destructors.md:417]: The tag here is supposed to be attached to the heading. When the content in the section is small enough to not require more than one of its own tags, I've just omitted the tags on the paragraphs themselves.

File: src/destructors.md [hunk 7]
```diff
@@ -256,9 +311,13 @@ always has the type `&'static Option<_>`, as it contains nothing disallowed).
 
 ### Temporary lifetime extension
 
+r[destructors.scope.lifetime-extension]
+
 > **Note**: The exact rules for temporary lifetime extension are subject to
 > change. This is describing the current behavior only.
 
+r[destructors.scope.lifetime-extension.let]
+
```
Comment (traviscross) [inline src/destructors.md:320]: Is there a reason for the line break here?

File: src/destructors.md (hunk 4 above)
Comment (traviscross) [reply src/destructors.md:417]: Makes sense.  Seems reasonable.

File: src/destructors.md (hunk 3 above)
Comment (chorman0773) [reply src/destructors.md:2]: I'd like to have the rendered syntax there, but link-check says no.

**Review by traviscross — APPROVED** [review]

**Pull request merged**